.github/workflows/external-pr.yml

```diff
@@ -0,0 +1,56 @@
+name: Build External PR
+
+on:
+  pull_request:
+    paths-ignore:
+      - '**.md'
+
+env:
+  # Base for the PR
+  BASE_REPO: ${{ github.event.pull_request.base.repo.clone_url }}
+
+  # Repository of PR
+  PR_HEAD_REPO: ${{ github.event.pull_request.head.repo.clone_url }}
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: ['ubuntu-latest', 'macos-latest' ] # 'windows-latest' too flaky
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: 'Check out repository'
+        uses: actions/checkout@v3
+        if: ${{ env.BASE_REPO != env.PR_HEAD_REPO }}
+
+      - name: 'Set up Java'
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: 'temurin'
+        if: ${{ env.BASE_REPO != env.PR_HEAD_REPO }}
+
+      - name: 'Cache Maven packages'
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml', '**/*.target') }}
+          restore-keys: ${{ runner.os }}-m2
+        if: ${{ env.BASE_REPO != env.PR_HEAD_REPO }}
```

Comment:
wonder if this can be moved up to build
https://blog.mergify.com/running-github-actions-only-on-certain-pull-requests/

Reply:
Unfortunately not, because of this https://github.com/actions/runner/issues/1189